### DISCUSSION TITLE
How can i get current image data when send to server

### DISCUSSION THREAD

**Discussion by ThorPham** [rpldy/react-uploady#352] [Q&A]
Thank for great library . I have a question.
I want to store image upload to useState .

**Comment by yoavniran** [comment]
Hi @ThorPham 
Will move this to the discussions area as this seems like a usage question.

Please do share more information on what exactly you're looking to achieve.

In short, you can use the [useItemStartListener](https://react-uploady.netlify.app/docs/api/hooks/useItemStartListener/) hook to do this. 

The [Upload Preview](https://react-uploady.netlify.app/docs/api/components/uploadPreview/) component may also be useful. Using a custom preview component. 

Without knowing more I can't really tell you what will be best.

**Comment by ThorPham** [comment]
@yoavniran Thank you for your reply.
I just fixed my issue. I want to visualize image before  upload and server response . So i want to store image original  , This is my code after search a lot of google. 
` useItemStartListener( (item)=>{
    var dataUrl = URL.createObjectURL(item.file)
    setImgSrc(dataUrl) ` }

**Comment by yoavniran** [comment]
@ThorPham if you want to show the image before/during upload you have Uploady's [Upload Preview](https://react-uploady.netlify.app/docs/api/components/uploadPreview/) component specifically for this use case. 

There's no need for you to write custom code for that. Of course, you're free to do so if you want but notice that the Preview component takes care of edge-cases and allows for customizations. 

This [sandbox](https://codesandbox.io/s/preview-progress-react-uploady-pki4i?file=/src/App.js) gives a good example of how to show preview with upload progress.

**Comment by ThorPham** [comment]
@yoavniran Oh my god. This is amazing example. I just learn reactjs and i don't know how to get image to render other component. Thank you so much.

**Comment by ThorPham** [comment]
@yoavniran I have problem . I want to show image upload to make background for canvas element. But i cann't find how to get image data.

**Comment by yoavniran** [comment]
The image data is provided to the PreviewComponent you can pass to the UploadPreview as a prop.
See the example in the docs: https://react-uploady.netlify.app/docs/api/components/uploadpreview/#custom-preview-component

**Comment by ThorPham** [comment]
@yoavniran but I don't want to show UploadPreview. Can i get data but not render UploadPreview ?

**Comment by yoavniran** [comment]
you can use [useItemStartListener](https://react-uploady.netlify.app/docs/api/hooks/useItemStartListener/) hook and call  [URL.createObjectURL](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL) on the `item.file` yourself
